spec/airport_spec.rb

```diff
@@ -1,48 +1,68 @@
 require 'airport'
+require 'plane'
+require 'byebug'
 
-## Note these are just some guidelines!
-## Feel free to write more tests!!
+describe Airport do
+  subject { Airport.new }
+  let(:plane) { Plane.new }
 
-# A plane currently in the airport can be requested to take off.
-#
-# No more planes can be added to the airport, if it's full.
-# It is up to you how many planes can land in the airport
-# and how that is implemented.
-#
-# If the airport is full then no planes can land
+  describe 'Airport can instruct take off' do
 
-describe Airport do
+    before(:each) do
+      allow(plane).to receive("land")
+      subject.landing(plane)
+      allow(plane).to receive("take_off")
+    end
 
-  describe 'take off' do
-    xit 'instructs a plane to take off'
+    it 'airport instructs a plane to take off' do
+      expect(subject).to respond_to(:take_off)
+    end
 
-    xit 'releases a plane'
+    it 'releases a plane' do
+      expect(subject.take_off).to eq(plane)
+    end
   end
 
-  describe 'landing' do
-    xit 'instructs a plane to land'
+  describe 'Airport can instruct a landing' do
+
+    before(:each) do
+      allow(plane).to receive(:land)
+    end
+
+    it 'airport instruct a plane to land' do
+      expect(subject).to respond_to(:landing)
+    end
 
-    xit 'receives a plane'
+    it 'airport receives a plane' do
+      expect(subject.landing plane).to eq(plane)
+    end
   end
 
-  describe 'traffic control' do
-    context 'when airport is full' do
-      xit 'does not allow a plane to land'
+  describe 'Airport Traffic controller' do
+    context 'airport is full' do
+      before(:each) do
+        allow(plane).to receive('land')
+      end
+
+      it 'traffic controller does not allow plane to land' do
+        subject.capacity.times { subject.landing(plane) }
+        expect { subject.landing(plane) }.to raise_error "Keep flying!"
+      end
     end
+  end
 
-    # Include a weather condition.
-    # The weather must be random and only have two states "sunny" or "stormy".
-    # Try and take off a plane, but if the weather is stormy,
-    # the plane can not take off and must remain in the airport.
-    #
-    # This will require stubbing to stop the random return of the weather.
-    # If the airport has a weather condition of stormy,
-    # the plane can not land, and must not be in the airport
+  describe 'Weather is sunny and stormy' do
+    context 'weather is to stormy' do
 
-    context 'when weather conditions are stormy' do
-      xit 'does not allow a plane to take off'
+      it 'Airport does not allow plane to take off' do
+        allow(subject).to receive(:weather).and_return(:stormy)
+        expect { plane.take_off(subject) }.to raise_error "You are not allowed to take off!"
```

Comment:
Metrics/LineLength: Line is too long. [92/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)